receiver/prometheusreceiver/metrics_receiver_report_extra_scrape_metrics_test.go

```diff
@@ -53,12 +54,14 @@ func testScraperMetrics(t *testing.T, targets []*testData, reportExtraScrapeMetr
 	require.NoErrorf(t, err, "Failed to create Prometheus config: %v", err)
 	defer mp.Close()
 
+	err = featuregate.GlobalRegistry().Set("receiver.prometheusreceiver.EnableReportExtraScrapeMetrics", reportExtraScrapeMetrics)
```

Comment:
In tests we try not to use the global registry. You can use this [helper function](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e053df555e761816861860f08fc12d00fb32dca4/internal/common/testutil/testutil.go#L132-L156) for that :)